### DISCUSSION TITLE
Improve topic search

### DISCUSSION THREAD

**Discussion by camchambers-cpr** [foxglove/community#913] [Q&A] [answered]
The results in the topic search seems to be limited by the size of the panel. 

It would be nice if it were wide enough to display everything or have a tooltip that displays the full topic.

![truncated](https://github.com/foxglove/community/assets/156825142/00eb4f2e-f614-4576-82d4-6b2a78d3d512)

**Comment by defunctzombie** [comment]
We've been thinking of moving this to the settings area to match what other panels typically do. Would that be an ok solution to this limitation?

**Reply by camchambers-cpr** [reply]
Wouldn't results still be truncated if the topic search was moved to the settings area?

![Screenshot from 2024-01-17 08-35-00](https://github.com/foxglove/community/assets/156825142/223724fa-0069-4b8f-acdb-b5810becad61)

**Reply by defunctzombie** [reply]
In the sidebar there's typically more space - and you can size it wider to work with longer topics. For the dropdown in the diagnostics titlebar the filter input takes up more space. I'm also looking at what we can do to remove clipping from the popup when using message path inputs on the sidebar.

**Comment by jasonimercer** [comment]
Moving the topic to a panel to match other workflows would be quite poor. The panel settings approach generally feels confusing to users as the settings and the results are decoupled. Here's an example of how webviz does it when the topics are wider than the available space.

![image](https://github.com/foxglove/community/assets/12039018/d2ddaa35-2c06-4605-b1e6-73a607fe4810)

**Comment by defunctzombie** [comment, marked as answer]
The original post screenshot of the panel seems outdated to me. We moved that setting to the settings area and generally put panel settings in that area.

The raw message panel is special and keeps its path selection in the panel. We recently improved the popup and it is wider than the input area and will look like the above screenshot.

**Reply by jtbandes** [reply]
Screenshot of current behavior:

<img width="378" alt="image" src="https://github.com/foxglove/community/assets/14237/1e59af46-abd9-42de-8b5b-d8026daba3a4">
